drf_haystack/serializers.py

```diff
@@ -131,24 +138,69 @@ def get_fields(self):
                     warnings.warn("Field '{field}' already exists in the field list. This *will* "
                                   "overwrite existing field '{field}'".format(field=field_name))
                 field_mapping[field_name] = declared_fields[field_name]
-
         return field_mapping
 
     def to_representation(self, instance):
         """
+        If we have a serializer mapping, use that.  Otherwise, use standard serializer behavior
+        """
+        serializers = getattr(self.Meta, "serializers", None)
+        if not serializers:
+            return self.default_to_representation(instance)
+
+        index = instance.searchindex
+        serializer_class = serializers.get(type(index), None)
+        if not serializer_class:
+            raise ImproperlyConfigured("Could not find serializer for %s in mapping" % index)
+        return serializer_class(context=self._context).to_representation(instance)
+
+    def default_to_representation(self, instance):
```

Comment:
Wouldn't it be more intuitive to let the `default_to_representation` code live in the default `to_representation` method, and the custom code to live in a custom method? Or even better if they could co-exist in the same method.
I had to read this twice to understand the flow here, and in my head; since the "custom code" is just a few lines, it could easily live in the same method.